src/main/java/entity_layer/TimeBlock.java

```diff
@@ -27,7 +27,8 @@ public interface TimeBlock {
     int hashCode();
```

Comment:
I don't think you need hashCode here because it is overridden from the object class